hooks/hooks.py

```diff
@@ -152,6 +156,23 @@ def install_etcd():
     hookenv.open_port(4001)
     db.set('installed', True)
 
+def isleader_available():
+    """Attempt to locate is_leader
+
+    predicate method to determine if we are on a Juju revision that maintains
+    the leader election codebase.
+    """
+    cmd = ['which', 'is_leader']
+    try:
+        ret = subprocess.call(cmd, universal_newlines=True)
+        if ret == 0:
+            return True
+        else:
+            return False
+    except OSError as e:
```

Comment:
I'd remove the entire try block here, since OSError will simply raise just like any other exception. There really isn't any exception handling here. You can also simplify the if block with

```
return ret == 0
```

Finally, it appears that [charmhelpers has some way of translating whether or not is-leader is available](http://bazaar.launchpad.net/~charm-helpers/charm-helpers/devel/view/head:/charmhelpers/core/hookenv.py#L800). Doing this

```
from charmhelpers.core.hookenv import is_leader

try:
    is_leader()
except NotImplementedError:
    print("panic message"
```

would achieve the same goal.